src/messages/responses/cache-dictionary-fetch.ts

```diff
@@ -0,0 +1,74 @@
+import {ResponseBase} from './response-base';
+import {SdkError} from '../../errors/errors';
+import {applyMixins, ErrorBody} from '../../errors/error-utils';
+import {TextDecoder} from 'util';
+import {cache_client} from '@gomomento/generated-types/dist/cacheclient';
+
+const TEXT_DECODER = new TextDecoder();
+
+export abstract class Response extends ResponseBase {}
+
+export class Hit extends Response {
+  private readonly items: cache_client._DictionaryFieldValuePair[];
+  private readonly dictionaryUint8ArrayUint8Array: Map<Uint8Array, Uint8Array> =
+    new Map();
+  private readonly dictionaryStringString: Map<string, string> = new Map();
+  private readonly dictionaryStringArrayBuffer: Map<string, Uint8Array> =
+    new Map();
+
+  constructor(items: cache_client._DictionaryFieldValuePair[]) {
+    super();
+    this.items = items;
+  }
+
+  public valueDictionaryUint8ArrayUint8Array(): Map<Uint8Array, Uint8Array> {
+    for (const item of this.items) {
+      this.dictionaryUint8ArrayUint8Array.set(item.field, item.value);
+    }
+    return this.dictionaryUint8ArrayUint8Array;
+  }
+
+  public valueDictionaryStringString(): Map<string, string> {
+    for (const item of this.items) {
+      this.dictionaryStringString.set(
+        TEXT_DECODER.decode(item.field),
+        TEXT_DECODER.decode(item.value)
+      );
+    }
+    return this.dictionaryStringString;
+  }
+
+  public valueDictionaryStringUint8Array(): Map<string, Uint8Array> {
+    for (const item of this.items) {
+      this.dictionaryStringArrayBuffer.set(
+        TEXT_DECODER.decode(item.field),
+        item.value
+      );
+    }
+    return this.dictionaryStringArrayBuffer;
+  }
+
+  public override toString(): string {
+    let stringRepresentation = '';
+    this.valueDictionaryStringString().forEach((value, key) => {
+      const keyValue = `${key}: ${value}, `;
+      stringRepresentation = stringRepresentation + keyValue;
+    });
+    return `${super.toString()}: valueDictionaryStringString: ${stringRepresentation.slice(
+      0,
+      -2
+    )}`;
```

Comment:
What's up with `slice(0, -2)`? Is this to remove the extra `, ` at the end?

Will this work with an empty dictionaryy?

might be worth writing a unit test for this toString() (don't have to cover all the cases, just super basic ones).

also might be worth grabbing the string truncate logic from Schwern's code (if applicable, i haven't put too much thoughts into this): https://github.com/momentohq/client-sdk-javascript/blob/0064bdece94623730362abccdaa48b5ff8b1e0d6/src/messages/responses/cache-list-fetch.ts#L39

Reply:
Using `slice(0, -2) on an empty string results in an empty string.
Even if `valueDictionaryStringString` returns an empty map, `toString()` will return an empty string (can be observed [here](https://jsfiddle.net/13s9vdeb/2/))

Reply:
There's code in listFetch that might be useful.